master/Master.hs

```diff
@@ -134,21 +138,21 @@ sendResponse kvMsg@(KVResponse txn_id slave_id _) = do
     -- forwarded to the client
     let oldAckMap = ackMap state
 
-    case Map.lookup txn_id oldAckMap of
+    case Map.lookup tid oldAckMap of
       Nothing -> do
-        let newAckMap = Map.insert txn_id (Set.singleton slave_id) oldAckMap
+        let newAckMap = Map.insert tid (Set.singleton sid) oldAckMap
 
         putMVar mvar $ state {ackMap = newAckMap} --MasterState (socket state) (cfg state) (voteMap state) newAckMap
 
         sendMsgToClient kvMsg (cfg state)
 
       Just hasRespondedSet  -> do
-        let hasRespondedSet' = Set.insert slave_id hasRespondedSet
+        let hasRespondedSet' = Set.insert sid hasRespondedSet
             newAckMap' = if Set.size hasRespondedSet' == (Prelude.length $ slaveConfig $ cfg state)
                          --erase this txn from the ack map, all the slaves have
                          --sent a response to master
-                         then Map.delete txn_id oldAckMap
-                         else Map.insert txn_id hasRespondedSet' oldAckMap
+                         then Map.delete tid oldAckMap
+                         else Map.insert tid hasRespondedSet' oldAckMap
 
         putMVar mvar $ state { ackMap = newAckMap'} --MasterState (socket state) (cfg state) (voteMap state) newAckMap'
 
```

Comment:
perhaps on line 163 we need
`updatedVoteMap    = Map.insert txn_id Set.empty (voteMap state)` ?